src/main/haskell/kore/src/Kore/Step/Function/Identifier.hs

```diff
@@ -0,0 +1,47 @@
+{-|
+Module      : Kore.Step.Function.Identifier
+Description : Data structures used for axiom evaluation.
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+Maintainer  : virgil.serbanuta@runtimeverification.com
+Stability   : experimental
+Portability : portable
+
+TODO(virgil): Move from Kore.Step.Function to Kore.Step.Axiom
+-}
+
+module Kore.Step.Function.Identifier
+    ( AxiomIdentifier (..)
+    , extract
+    ) where
+
+import Kore.AST.Common
+       ( SymbolOrAlias (..) )
+import Kore.AST.Identifier
+       ( Id (..) )
+import Kore.AST.Pure
+       ( PurePattern )
+import Kore.AST.Valid
+       ( pattern App_, pattern Ceil_ )
+
+data AxiomIdentifier level
+    = Application !(Id level)
+    | Ceil !(AxiomIdentifier level)
+    deriving (Eq, Ord, Show)
+
+extract
```

Comment:
Could we name this `extractAxiomIdentifier` to avoid colliding with `Control.Comonad.extract`? Alternative, to this and the comment above: could we add documentation to this module indicating it is intended to be imported qualified?